pkg/argocd/argocd.go

```diff
@@ -0,0 +1,239 @@
+package argocd
+
+import (
+	"context"
+	"fmt"
+
+	argocdoperatorv1alpha1 "github.com/argoproj-labs/argocd-operator/api/v1alpha1"
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides struct for the argocd object containing connection to
+// the cluster and the argocd definitions.
+type Builder struct {
+	// argocd Definition, used to create the argocd object.
+	Definition *argocdoperatorv1alpha1.ArgoCD
+	// created argocd object.
+	Object *argocdoperatorv1alpha1.ArgoCD
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// used to store latest error message upon defining the argocd definition.
+	errorMsg string
+}
+
+// AdditionalOptions additional options for argocd object.
+type AdditionalOptions func(builder *Builder) (*Builder, error)
```

Comment:
Please remove it. You are not using it in mutation functions.

Reply:
Done.